src/components/cdPipeline/CDPipeline.tsx

```diff
@@ -283,15 +292,65 @@ export default function CDPipeline({
         setAvailableTags(tags)
     }
 
-    useEffect(() => {
-        getInit()
-    }, [])
+    const handlePopulatePluginDataStore = async (form: PipelineFormType) => {
+        if (!form) {
+            return
+        }
 
-    useEffect(() => {
-        if (formData.environmentId) {
-            getConfigMapSecrets()
+        const preBuildPluginIds = getPluginIdsFromBuildStage(form.preBuildStage)
+        const postBuildPluginIds = getPluginIdsFromBuildStage(form.postBuildStage)
+        const uniquePluginIds = Array.from(new Set([...preBuildPluginIds, ...postBuildPluginIds]))
+
+        if (processPluginData) {
+            await getMandatoryPluginData(form, uniquePluginIds)
+            return
+        }
+
+        if (uniquePluginIds.length === 0) {
+            return
         }
-    }, [formData.environmentId])
+
+        const {
+            pluginStore: { parentPluginStore, pluginVersionStore },
+        } = await getPluginsDetail({
+            appId: +appId,
+            pluginIds: uniquePluginIds,
+            shouldShowError: false,
+        })
+
+        handlePluginDataStoreUpdate(getUpdatedPluginStore(pluginDataStore, parentPluginStore, pluginVersionStore))
+    }
+
+    const getEnvCDPipelineName = (form) => {
+        Promise.all([getCDPipelineNameSuggestion(appId), getEnvironmentListMinPublic(true)])
+            .then(([cpPipelineName, envList]) => {
+                form.name = cpPipelineName.result
+                let list = envList.result || []
+                list = list.map((env) => {
+                    return {
+                        id: env.id,
+                        clusterId: env.cluster_id,
+                        clusterName: env.cluster_name,
+                        name: env.environment_name,
+                        namespace: env.namespace || '',
+                        active: false,
+                        isClusterCdActive: env.isClusterCdActive,
+                        description: env.description,
+                        isVirtualEnvironment: env.isVirtualEnvironment,
+                        allowedDeploymentTypes: env.allowedDeploymentTypes || [],
+                        isDigestEnforcedForEnv: env.isDigestEnforcedForEnv,
+                    }
+                })
+                sortObjectArrayAlphabetically(list, 'name')
```

Comment:
suggestion: potential scope for clean up

Reply:
https://github.com/devtron-labs/dashboard/pull/2131#discussion_r1811083924